client_type.go

```diff
@@ -55,3 +55,6 @@ type RetryStrategyFunc func(int64) time.Duration
 
 // IterateItemFunc represents the callback to iterate a list of results, if false stops
 type IterateItemFunc func(interface{}, error) bool
+
+// WaitAsyncJobResultFunc represents the callback to wait a results of an async request, if false stops
+type WaitAsyncJobResultFunc func(interface{}, interface{}, error) bool
```

Comment:
`func(JobStatus, interface{}, error)` ?

Reply:
or we find a way to put everything into `JobResult` and postpone the JSON unmarshalling

```go
jobResult.JobStatus
jobResult.Result().(*egoscale.VirtualMachine)
```

Reply:
Yes it's really better like that !

Reply:
```golang
vm := &VirtualMachine{}
if err := asyncJobResult.Result(vm); err {
    ...
}
```